.github/workflows/build-clang-image.yaml

```diff
@@ -0,0 +1,96 @@
+name: Build Clang Image
+
+on:
+  push:
+    branches:
+      - main
+      - v*
+    paths:
+    - 'Dockerfile.clang'
+  pull_request:
+    paths:
+    - 'Dockerfile.clang'
+
+jobs:
+  build-and-push:
+    environment: release-clang
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Login to quay.io
+        uses: docker/login-action@49ed152c8eca782a232dede0303416e8f356c37b
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_CLANG_RELEASE_USERNAME }}
+          password: ${{ secrets.QUAY_CLANG_RELEASE_PASSWORD }}
+
+      - name: Getting image tag
+        id: tag
+        run: |
+          if [ ${{ github.event.pull_request.head.sha }} != "" ]; then
+            echo ::set-output name=tag::${{ github.event.pull_request.head.sha }}
+          else
+            echo ::set-output name=tag::${{ github.sha }}
+          fi
+
+      - name: Checkout Source Code
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Release Build clang
+        uses: docker/build-push-action@e551b19e49efd4e98792db7592c17c09b89db8d8
+        id: docker_build_release
+        with:
+          context: .
+          file: ./Dockerfile.clang
+          push: true
+          platforms: linux/amd64
+          tags: |
+            quay.io/${{ github.repository_owner }}/clang:${{ steps.tag.outputs.tag }}
```

Comment:
Should we also tag  based on clang version?

Reply:
ok, I pushed new version with this.. please check

Reply:
not sure about this one.. @willfindlay ?

Reply:
@kkourt something like `clang:${version}-latest`? yeah, that sounds reasonable to me.

Reply:
Yes, something like that. My thinking was that it would be nice it would be easier when trying out things to get the (latest) clang 13 or clang 14 version. Not a big deal though, and we can always do it later if needed.